lib/alarm_service/screens/ring.dart

```diff
@@ -13,29 +15,27 @@ class AlarmRingScreen extends StatefulWidget {
 }
 
 class _AlarmRingScreenState extends State<AlarmRingScreen> {
+  static final _log = Logger('ExampleAlarmRingScreenState');
+
+  StreamSubscription<AlarmSet>? _ringingSubscription;
+
   @override
   void initState() {
     super.initState();
-    Timer.periodic(const Duration(seconds: 1), (timer) async {
-      if (!mounted) {
-        timer.cancel();
-        return;
-      }
-
-      final isRinging = await Alarm.isRinging(widget.alarmSettings.id);
-      if (isRinging) {
-        alarmPrint('Alarm ${widget.alarmSettings.id} is still ringing...');
-        return;
-      }
-
-      alarmPrint('Alarm ${widget.alarmSettings.id} stopped ringing.');
-      timer.cancel();
-      if (mounted) {
-        Navigator.pop(context);
-      }
+      _ringingSubscription = Alarm.ringing.listen((alarms) {
+      if (alarms.containsId(widget.alarmSettings.id)) return;
+      _log.info('Alarm ${widget.alarmSettings.id} stopped ringing.');
+      _ringingSubscription?.cancel();
+      if (mounted) Navigator.pop(context);
```

Comment:
```suggestion
      _ringingSubscription = Alarm.ringing.listen((alarms) {
      if (alarms.containsId(widget.alarmSettings.id)) {
        return;
      }
      _log.info('Alarm ${widget.alarmSettings.id} stopped ringing.');
      _ringingSubscription?.cancel();
      if (mounted) {
        Navigator.pop(context);
      }
```

Nekem ez a kódstílus átláthatóbb, de el tudom engedni, ha neked meg az egy soros jobb.